enhancements/windows-containers/windows-machine-config-operator-upgrades.md

```diff
@@ -0,0 +1,206 @@
+---
+title: windows-machine-config-operator-upgrades
+authors:
+  - "@sebsoto"
+reviewers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+approvers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+creation-date: 2020-07-10
+last-updated: 2020-07-10
+status: implementable
+---
+
+# Windows Machine Config Operator Upgrades
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement allows a cluster administrator to upgrade a deployed Windows
+Machine Config Operator (WMCO), and associated Windows node components, to a
+newer version.
+
+## Motivation
+
+The main motivation behind this enhancement is to give the end user
+an easy and non-disruptive way to upgrade to a newer release of the WMCO, and
+upgrade associated Windows node components with it.
+WMCO is not part of the release payload, and is managed with [OLM](https://github.com/operator-framework/operator-lifecycle-manager).
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Enable upgrading WMCO with as little disruption as possible.
+* Enable upgrading on-node Kubernetes components through newer versions of the
+  WMCO, including major version upgrades across OpenShift cluster upgrades, and
+  bug and CVE fixes during the current version.
+
+### Non-Goals
+
+In this enhancement we do not plan to support Windows operating system updates.
+
+## Proposal
+
+To allow for the WMCO to be upgraded in a safe and non-disruptive way, it is
+proposed that Windows VMs configured by previous versions of the WMCO are
+indirectly drained and terminated in batches by user created MachineHealthChecks,
+triggered by WMCO stopping the associated node's kubelet. The upgraded WMCO
+instance will then be able to configure VMs that will be created in place of
+the terminated ones.
+
+In order to allow for this to occur with minimal service disruption, it is
+recommended that users have at least two Windows nodes within the cluster.
+
+All Windows Kubernetes component updates will be tied to WMCO releases.
+
+### Justification
```

Comment:
Implicit in this design is you upgrade the operator, but not the nodes.

It would be good to link to the restrictions of windows containers that make any desire to do in-place upgrade not helpful for the application.

For example, since you cannot project volumes from host storage, there is no node local data to worry about preserving: https://kubernetes.io/docs/setup/production-environment/windows/intro-windows-in-kubernetes/#storage

Reply:
Thanks for the the suggestion, I have added the example you've provided to the doc